internal/uploader/uploader.go

```diff
@@ -0,0 +1,28 @@
+package uploader
+
+import (
+	"context"
+
+	"github.com/go-openapi/strfmt"
+	eventsapi "github.com/openshift/assisted-service/internal/events/api"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type UploadHandler interface {
```

Comment:
Maybe just `Client` since `uploader` is already the package name?

Then the callers would do something like:
```go
dataUploadClient := uploader.NewClient(...)
```

Reply:
That definitely sounds a lot better! Modified it to `Client`, how does it look?